### ISSUE TITLE
C stack usage too close to limit

### ISSUE THREAD

**Issue by jsicherman** [gaborcsardi/async#65]
Hi,

When running this package, I will sometimes have my session freeze up for some minutes and then resume with an `Error: C stack usage  XXXXXXX is too close to the limit` (both while code is running and not). My naive assumption is that it's caused by my nested `http_get` calls (ie. below). Perhaps the event manager is trying to release the responses and there is some infinite crosstalk until it hits my C stack limit?

```
http_get('...')$then(function(response) {
    http_get('...')$then(function(response2) {
        foo(response$content, response2$content)
    })
})
```

I can't replicate this error reliably. Wondering if you've run into something similar, or if I should be simply catching the `response$content` and forcing a `gc` of `response` before making the second `http_get`? Output from `Cstack_info()` below.

```
> Cstack_info()
      size    current  direction eval_depth 
   7969177      23232          1          2 
```

**Comment by lionel-** [comment]
I ran into this as well, I hoped that #62 would fix it.

**Comment by jsicherman** [comment]
To clarify, did #63 fix it for you?

**Comment by gaborcsardi** [comment]
Yes #63 fixes #62, there is a test case for it: 
https://github.com/gaborcsardi/async/blob/3559f423d3758e16bbe0eee1df67da0d43c1d0cc/tests/testthat/test-deferred.R#L71

Your issue seems the same.

**Comment by jsicherman** [comment]
Updating to the current GitHub version seems to have fixed. Thanks!
